src/transformers/models/clip/modeling_clip.py

```diff
@@ -950,7 +1043,9 @@ def __init__(self, config: CLIPConfig):
             )
 
         text_config = config.text_config
+        text_config._attn_implementation = config._attn_implementation
         vision_config = config.vision_config
+        vision_config._attn_implementation = config._attn_implementation
```

Comment:
The image classification class has:

https://github.com/huggingface/transformers/blob/8b02bb6e748842cfb74ec9cba808201b893fa0af/src/transformers/models/clip/modeling_clip.py#L1332

If we don't propagate `vision_config._attn_implementation = config._attn_implementation` the vision config won't have any way to know about the actual `_attn_implementation`. 

However, correct me if I am wrong.

Reply:
You're completely right! However, the way it's propogated in other models is by passing in the model construction e.g. 

```py
self.text_model = CLIPTextTransformer.from_config(text_config, attn_implementation=config._attn_implementation)
```

I'd encourage doing it this way for two reasons: 
* It's the pattern done for other models. If there's an issue we need to update in the code, we're more likely to find it if it matches
* The way the attention implementation is set on the config is less than ideal and (imo) prone to unexpected behaviour. Annoyingly, there's a bunch of magic which happens in the setter, which can cause it to appear to magically change or revert back. This way I'm sure (atm) works, I'm not sure that setting on the config like this and then passing to the model will always work, depending on the attention implementation the original config e.g. `vision_config`

Reply:
Done in 89dab66fa. Keeping this comment open because it seems like an important thing for us to consider.